titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/volume/Volume.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api.jobmanager.model.job.volume;
+
+import javax.validation.Valid;
+
+public class Volume {
+    @Valid
+    private static String name;
+
+    public Volume(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Builder toBuilder() {
+        return newBuilder();
+    }
+
+    public static Builder newBuilder() {
+        return new Builder(name);
+    }
+
+    public static class Builder {
```

Comment:
To avoid casting when using this builder it should include template parameter.

Reply:
I don't understand the java type system to know exactly what those words mean :(
Let me know on the next review if it is still relevant

Reply:
Ok I think I understand, but was this comment with regards to VolumeSource and not Volume? 
Because I think the VolumeSource stuff is where I have to do the casting.